.gitlab-ci.yml

```
@@ -26,23 +28,41 @@ variables:
     --time=00:10:00 
     --account=SHELLARD-SL3-GPU 
     --partition=ampere
+  FCOMPARE_FLAGS: >
+    --abs_tol 1e-10
+    --rel_tol 1e-10
+    --abort_if_not_all_found
+  LOCKED_FILENAME: >
+    file.lock
+
 
 csd3-a100:
   stage: test
   tags:
     - csd3
   script:
+    - module purge
+    - module use ${MODULEPATHS_A100}
+    - module load ${MODULES_A100}
     - if [[ -d ../amrex ]]; then
     -   cd ../amrex
     -   git fetch --depth 1
     -   git reset --hard origin/development
     - else
     -   git clone --depth 1 https://github.com/AMReX-Codes/amrex.git ../amrex
     - fi
+    - AMREX_HOME="$PWD"
+    - cd ${AMREX_HOME}/Tools/Plotfile
+    - make -j 8 COMP=gnu programs=fcompare
     - cd ${HOME}/${CI_PROJECT_DIR}/Tests
-    - module purge
-    - module use ${MODULEPATHS_A100}
-    - module load ${MODULES_A100}
-    - make -j 8 ${BUILD_CONFIG}
-    - srun ${SRUN_FLAGS} make run ${BUILD_CONFIG}
+    - make -j 8 ${BUILD_CONFIG} 
+    - flock ${HOME}/${CI_PROJECT_DIR}/${LOCKED_FILENAME} srun ${SRUN_FLAGS} make run ${BUILD_CONFIG}
```

I think we want this to be somewhere common to all jobs so we want to remove the dependence on `${CI_PROJECT_DIR}`:
```suggestion
    - flock ${HOME}/${QOS_INTR_LOCK_FILE} srun ${SRUN_FLAGS} make run ${BUILD_CONFIG}
```